doc/api/assert.md

```diff
@@ -916,20 +919,24 @@ assert(0);
 <!-- YAML
 added: REPLACEME
 -->
-* `block` {Function}
-* `error` {RegExp|Function|Object}
+* `block` {Function|Promise}
+* `error` {RegExp|Function|Object|Error}
 * `message` {any}
 
-Awaits for promise returned by function `block` to be rejected.
+Awaits the `block` promise or, if `block` is a function, immediately call the
+function and await the returned promise to complete. It will then check that the
+promise is rejected.
 
-When `assert.rejects()` is called, it will immediately call the `block`
-function, and awaits for completion.
+If `block` is a function and it throws synchronously, that error will be
+rejected directly without checking the `error` handler.
```

Comment:
what do you mean with this?

Reply:
Synchronously thrown errors (due to the `block()` call) will directly cause a rejected promise to be returned (with that error). That is different in the way that a rejection caused by calling `block()` will trigger the potential `error` handler (second argument). If the rejected error does not adhere to the error handler, then it will cause a rejection. Otherwise not.

I do see that this is not well worded yet and it would be great to get a suggestion for improvement. Looping in @nodejs/documentation

Reply:
Ping @vsemozhetbyt @Trott

Reply:
I cannot think of anything more clear(

Reply:
Maybe something like this?

> If `block` is a function and it throws an error synchronously, `assert.rejects` will return a rejected Promise with that error without checking the `error` handler.